storage/src/device.rs

```diff
@@ -652,6 +654,38 @@ impl BlobIoVec {
     }
 }
 
+/// Helper structure to merge blob IOs to reduce IO requests.
+#[derive(Default)]
+pub struct BlobIoMerge {
+    map: HashMap<String, BlobIoVec>,
+    current: String,
+}
```

Comment:
I don't understand how the prefetch merge is improved with this patch. Could you help explain further?
Besides, it never tries to merge backend blob io from Rafs layer. It only gathers chunks that might be on different blobs and issue them to the storage layer. Backend blob IO will be merged in the storage layer.  Moreover, seems the newly introduced structure `BlobIoMerge` brings many memory allocation/overhead. Afterwards the hash map key is never used when issue prefetch.

Reply:
The rafs layer prefetch will try to prefetch one or more files. We have observed that those merged requests are still small, due to adjacent files are stored on different blobs. So the patch tries to merge
```
file 1: blob 1, chunk 1
file 2: blob 2, chunk 1
file 3: blob 1, chunk 2
file 4: blob 2, chunk 2
```
into
```
request 1: blob 1, chunk 1-2
request 2: blob 2, chunk 1-2
```
With current implementation, four requests will be issued instead of two.

Reply:
And compared with network latency, a few more memory allocation are acceptable tradeoff:)